src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -539,6 +569,32 @@ export class StudyViewPageStore
             )
         );
 
+        this.reactionDisposers.push(
+            reaction(
+                () => this.molecularProfileNameSet.isComplete,
+                isComplete => {
+                    if (isComplete) {
+                        const nameSet = this.molecularProfileNameSet.result!;
+                        for (const profileId of Object.keys(nameSet)) {
+                            const profileName = nameSet[profileId];
+                            const chartMeta = {
+                                uniqueKey: getGenePanelChartKey(profileId),
+                                displayName: 'Gene Panels: ' + profileName,
+                                description: '',
+                                chartType: ChartTypeEnum.PIE_CHART,
+                                dataType: ChartMetaDataTypeEnum.GENOMIC,
+                                patientAttribute: false,
+                                dimension: { w: 1, h: 1 },
+                                priority: 999,
+                                renderWhenDataChange: false,
+                            };
+                            this.setCustomChart(chartMeta);
```

Comment:
since we are calling this inside a loop and it mutates state, we need to use mobx "action" to update state, so that we don't get multiple update cycles.  easiest thing to do would be wrap in mobx runInAction.  alternative: allow setCustomChart to accept multiple chartMeta.  This is a little nicer because it's only one level of "action"